### PR TITLE
docs: add theory from v2 paper

### PR DESCRIPTION
**Pull request by njzjz**
Add theory to documentation.

### CODE REVIEW

**Comment by njzjz** [comment]
CI is fixed in #2950

File: doc/model/train-se-e2-a.md [hunk 1]
```diff
@@ -4,6 +4,59 @@ The notation of `se_e2_a` is short for the Deep Potential Smooth Edition (DeepPo
...
+    \mathcal{D}^i = \frac{1}{N_c^2} (\mathcal{G}^i)^T \mathcal{R}^i (\mathcal{R}^i)^T \mathcal{G}^i_<,
+```
+
+where
+$N_c$ is the expected maximum number of neighboring atoms, which is the same constant for all atoms over all frames.
+A matrix with a dimension of $N_c$ will be padded if the number of neighboring atoms is less than $N_c$.$\mathcal{R}^i \in \mathbb{R}^{N_c \times 4}$ is the coordinate matrix, and each row of $\mathcal{R}^i$ can be constructed as
+
+```math
+    (\mathcal{R}^i)_j =
+    \begin{array}{cccc}
```
Comment (Yi-FanLi) [inline doc/model/train-se-e2-a.md:24]: Is \{ missing before this line?

File: doc/model/train-se-e2-r.md [hunk 2]
```diff
@@ -2,6 +2,56 @@
...
+
+The descriptor, using either radial-only information, is given by
+
+```math
+    \mathcal{D}^i = \frac{1}{N_c} \sum_j (\mathcal{G}^i)_{jk},
+```
+
+where
+$N_c$ is the expected maximum number of neighboring atoms, which is the same constant for all atoms over all frames.
+A matrix with a dimension of $N_c$ will be padded if the number of neighboring atoms is less than $N_c$.$\mathcal{R}^i \in \mathbb{R}^{N_c}$ is the coordinate matrix, and each row of $\mathcal{R}^i$ can be constructed as
```
Comment (Yi-FanLi) [inline doc/model/train-se-e2-r.md:15]: ... is less than $N_c$.$\mathcal{R}^i \in ...
There should be a space between the period and the $\mathcal{R}.

File: doc/model/train-se-e2-a.md (hunk 1 above)
Comment (njzjz) [reply doc/model/train-se-e2-a.md:24]: Agree.
```suggestion
    \{
    \begin{array}{cccc}
```

File: doc/model/train-se-e3.md [hunk 3]
```diff
@@ -2,6 +2,37 @@
...
+```math
+    \mathcal{D}^i = \frac{1}{N_c^2}(\mathcal{R}^i(\mathcal{R}^i)^T):\mathcal{G}^i,
+```
+where
+$N_c$ is the expected maximum number of neighboring atoms, which is the same constant for all atoms over all frames.
+$\mathcal{R}^i$ is constructed as
+
+```math
+    (\mathcal{R}^i)_j =
+    \begin{array}{cccc}
```
Comment (njzjz) [inline doc/model/train-se-e3.md:17]:
```suggestion
    \{
    \begin{array}{cccc}
```

File: doc/model/train-se-e2-r.md (hunk 2 above)
Comment (njzjz) [reply doc/model/train-se-e2-r.md:15]:
```suggestion
A matrix with a dimension of $N_c$ will be padded if the number of neighboring atoms is less than $N_c$. $\mathcal{R}^i \in \mathbb{R}^{N_c}$ is the coordinate matrix, and each row of $\mathcal{R}^i$ can be constructed as
```

File: doc/model/train-se-e2-a.md [hunk 4]
```diff
@@ -4,6 +4,59 @@ The notation of `se_e2_a` is short for the Deep Potential Smooth Edition (DeepPo
...
+We call it two-body embedding because the embedding network takes only the distance between atoms $i$ and $j$ (see below), but it is not implied that the descriptor takes only the pairwise information between $i$ and its neighbors.
+The descriptor, using full information, is given by
+
+```math
+    \mathcal{D}^i = \frac{1}{N_c^2} (\mathcal{G}^i)^T \mathcal{R}^i (\mathcal{R}^i)^T \mathcal{G}^i_<,
+```
+
+where
+$N_c$ is the expected maximum number of neighboring atoms, which is the same constant for all atoms over all frames.
+A matrix with a dimension of $N_c$ will be padded if the number of neighboring atoms is less than $N_c$.$\mathcal{R}^i \in \mathbb{R}^{N_c \times 4}$ is the coordinate matrix, and each row of $\mathcal{R}^i$ can be constructed as
```
Comment (njzjz) [inline doc/model/train-se-e2-a.md:20]:
```suggestion
A matrix with a dimension of $N_c$ will be padded if the number of neighboring atoms is less than $N_c$. $\mathcal{R}^i \in \mathbb{R}^{N_c \times 4}$ is the coordinate matrix, and each row of $\mathcal{R}^i$ can be constructed as
```

**Review by wanghan-iapcm — APPROVED** [review]

File: doc/model/train-energy.md [hunk 5]
```diff
@@ -2,6 +2,62 @@
...
+where $F_{k,\alpha}$ is the $\alpha$-th component of the force on atom $k$, and the superscript $\ast$ indicates the label of the property that should be provided in advance.
+Using $N$ ensures that each loss of fitting property is averaged over atomic contributions before they contribute to the total loss by weight.
+
+If part of atoms is more important than others, for example, certain atoms play an essential role when calculating free energy profiles or kinetic isotope effects, the MSE of atomic forces with prefactors $q_{k}$ can also be used as the loss function:
+```math
+    L_F^p(\mathbf{x};\boldsymbol{\theta})=\frac{1}{3N}\sum_{k=1}^{N} \sum_{\alpha} q_{k} (F_{k,\alpha}(\mathbf{x};\boldsymbol{\theta})-F_{k,\alpha}^*)^2.
+```
+The atomic forces with larger prefactors will be fitted more accurately than those in other atoms.
+
+If some forces are quite large, for example, forces can be greater than 60~eV/\AA{} in high-temperature reactive simulations, one may also prefer the force loss is relative to the magnitude:
```
Comment (Yi-FanLi) [inline doc/model/train-energy.md:51]: 60 eV/\\AA?

File: doc/model/train-se-e2-a-tebd.md [hunk 6]
```diff
@@ -4,7 +4,59 @@ We generate specific a type embedding vector for each atom type so that we can s
...
+Take an example, the atomic energy $E_i$ is represented as follows:
+```math
+E_i=\mathcal{F}_0^{\alpha_i}(\mathcal{D}^i).
+```
+
+To reduce the number of NN parameters and improve computing efficiency when there are large numbers of chemical species,
+the type embedding $\mathcal{A}$ is introduced, represented as a NN function $\mathcal{N}_t$ of the atomic type $\alpha$:
+
+```math
+    \mathcal{A}^i = \mathcal{N}_t\big( \text{one\_hot}(\alpha_i) \big),
```
Comment (Yi-FanLi) [inline doc/model/train-se-e2-a-tebd.md:35]: one\_hot does not display correctly in the output document.

File: doc/model/train-se-e2-a-tebd.md [hunk 7]
```diff
@@ -4,7 +4,59 @@ We generate specific a type embedding vector for each atom type so that we can s
...
+```
+
+where $\alpha_i$ is converted to a one-hot vector representing the chemical species before feeding to the NN.
+The NN function \del{will be introduced in Section~\ref{section:NN}}\add{$\mathcal{N}$ was given in Eq.~\eqref{eq:NN}}.
+Based on Eqs.~\eqref{eq:G2} and \eqref{eq:nn-descrpt-e3}, the type embeddings of central and neighboring atoms $\mathcal{A}^i$ and $\mathcal{A}^j$ are added as an extra input of the embedding network $\mathcal{N}_{e,\{2,3\}}$:
```
Comment (Yi-FanLi) [inline doc/model/train-se-e2-a-tebd.md:40]: The eqrefs should be fixed.

**Review by Yi-FanLi — APPROVED** [review]
I have pointed out some typographical errors. Once you correct those small errors, this PR looks good to me.

File: doc/model/train-se-e3.md [hunk 8]
```diff
@@ -2,6 +2,37 @@
 
 The notation of `se_e3` is short for the Deep Potential Smooth Edition (DeepPot-SE) constructed from all information (both angular and radial) of atomic configurations. The embedding takes angles between two neighboring atoms as input (denoted by `e3`).
```
[the remark this answers is not in the document]
Comment (njzjz) [reply doc/model/train-se-e3.md:3]:
```suggestion
The notation of `se_e3` is short for the Deep Potential Smooth Edition (DeepPot-SE) constructed from all information (both angular and radial) of atomic configurations. The embedding takes bond angles between a central atom and its two neighboring atoms as input (denoted by `e3`).
```

**Pull request merged**